File: src/main/java/ru/romangr/catbot/handler/UnknownCommandHandler.java
package ru.romangr.catbot.handler;

import lombok.RequiredArgsConstructor;
import ru.romangr.exceptional.Exceptional;
import ru.romangr.catbot.executor.action.TelegramAction;
import ru.romangr.catbot.executor.action.TelegramActionFactory;
import ru.romangr.catbot.telegram.model.Chat;

import java.util.List;

@RequiredArgsConstructor
public class UnknownCommandHandler {

    private final TelegramActionFactory actionFactory;

    public Exceptional<HandlingResult> handle(Chat chat) {
        TelegramAction action
                = actionFactory.newSendMessageAction(chat, "Incorrect command syntax");
        return Exceptional.exceptional(HandlingResult.builder()
                .status(HandlingStatus.HANDLED)
                .actions(List.of(action))
                .build());
    }

}

File: src/main/java/ru/romangr/catbot/handler/UnknownCommandHandler.kt
package ru.romangr.catbot.handler

import ru.romangr.catbot.executor.action.TelegramActionFactory
import ru.romangr.catbot.telegram.model.Chat
import ru.romangr.exceptional.Exceptional

class UnknownCommandHandler(private val actionFactory: TelegramActionFactory) {

    fun handle(chat: Chat): Exceptional<HandlingResult> {
        val action = actionFactory.newSendMessageAction(chat, "Incorrect command syntax")
        return Exceptional.exceptional(HandlingResult(listOf(action), HandlingStatus.HANDLED))
    }

}
